recipes/_hadoop_hdfs_ha_checkconfig.rb

```diff
@@ -27,6 +27,9 @@
 # We have dfs.nameservices, and now need to check them
 dfs_nameservices.each do |ns|
   # Start namenode checks
+  if node['hadoop']['hdfs_site']['dfs.ha.namenodes'] 
+    Chef::Application.fatal!("NameNode HA requires node['hadoop']['hdfs_site']['dfs.ha.namenodes.#{ns}']") unless node['hadoop']['hdfs_site'].key?("dfs\.ha\.namenodes\.#{ns}")
+  end
```

Comment:
Probably, do you see a logic issue though?

I've intended to do ha by fulfilling this
if node['hadoop'].key?('hdfs_site') &&
node['hadoop']['hdfs_site'].key?('dfs.nameservices')


but the required settings are getting skipped because of this line
next unless node['hadoop']['hdfs_site'].key?("dfs\.ha\.namenodes\.#{ns}")

You need to set
dfs.ha.namenodes.nameservice
which I did not and it converged without error and without the required
values.





On Thu, Jul 2, 2015 at 4:28 PM, Chris Gianelloni <notifications@github.com>
wrote:

> In recipes/_hadoop_hdfs_ha_checkconfig.rb
> <https://github.com/caskdata/hadoop_cookbook/pull/209#discussion_r33819329>
> :
>
> > @@ -27,6 +27,9 @@
> >  # We have dfs.nameservices, and now need to check them
> >  dfs_nameservices.each do |ns|
> >    # Start namenode checks
> > +  if node['hadoop']['hdfs_site']['dfs.ha.namenodes']
> > +    Chef::Application.fatal!("NameNode HA requires node['hadoop']['hdfs_site']['dfs.ha.namenodes.#{ns}']") unless node['hadoop']['hdfs_site'].key?("dfs\.ha\.namenodes\.#{ns}")
> > +  end
>
> That was really confusing to read. Anyway, there's no such property
> dfs.ha.namenodes in hdfs-site.xml which means your if statement should
> never be matched. It makes your entire code block pointless.
>
>
> https://hadoop.apache.org/docs/r2.4.1/hadoop-project-dist/hadoop-hdfs/hdfs-default.xml
>
> It sounds like you're attempting to setup HA without actually configuring
> all of the required options, but I really cannot tell.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/caskdata/hadoop_cookbook/pull/209/files#r33819329>.
>